### PR TITLE
[WIP] Amazon SES

### PR DESCRIPTION
**Pull request by mitchbregs**

### CODE REVIEW

File: betfund_solicitor/ses.py [hunk 1]
```diff
@@ -1 +1,83 @@
...
+        payload = self.message.ses_send_email_payload
+
+        try:
+            # try to send the email
+            response = self.client.send_email(**payload)
+        except ClientError as e:
+            # raise error on connectivity or payload failure
+            raise e
+
+        return response
```
Comment (leonkozlowski) [inline betfund_solicitor/ses.py]: newline

File: betfund_solicitor/ses.py [hunk 2]
```diff
@@ -1 +1,83 @@
...
+            aws_secret_access_key=aws_secret_key,
+            region_name=aws_region
+        )
+        self.message = message
+
+    def __repr__(self):
+        return f"<SimpleEmailService ({self.message})>"
+
+    def send_email(self):
+        """Sends email using AWS `boto3.client.send_email` for SES."""
```
Comment (leonkozlowski) [inline betfund_solicitor/ses.py]: Returns in docstring

File: betfund_solicitor/ses.py [hunk 3]
```diff
@@ -1 +1,83 @@
 # See: https://docs.aws.amazon.com/ses/latest/DeveloperGuide/send-using-sdk-python.html
+
+from boto3 import client
+from botocore.exceptions import ClientError
+
+from betfund_solicitor import Message
+
+AWS_REGIONS = {
```
Comment (leonkozlowski) [inline betfund_solicitor/ses.py]: Move this to a config / or somewhere external

File: betfund_solicitor/ses.py [hunk 4]
```diff
@@ -1 +1,83 @@
...
+        """Sends email using AWS `boto3.client.send_email` for SES."""
+
+        payload = self.message.ses_send_email_payload
+
+        try:
+            # try to send the email
+            response = self.client.send_email(**payload)
+        except ClientError as e:
+            # raise error on connectivity or payload failure
+            raise e
```
Comment (leonkozlowski) [inline betfund_solicitor/ses.py]: consider changing `e` to some more verbose

**Review by leonkozlowski — CHANGES_REQUESTED** [review]
Mostly cosmetic, look into fixing the `pylint` issues as well (looks like deps are not installed correctly)

File: betfund_solicitor/ses.py [hunk 5]
```diff
@@ -1 +1,83 @@
...
+        AWS Region used for Amazon SES.
+    """
+    def __init__(
+        self,
+        message: Message,
+        aws_access_key: str,
+        aws_secret_key: str,
+        aws_region: str = "us-east-1"
+    ):
+        assert aws_region in AWS_REGIONS, 'not a valid `aws_region`'
```
Comment (leonkozlowski) [inline betfund_solicitor/ses.py]: IMO assertions are not graceful failures

**Comment by mitchbregs** [comment]
> Mostly cosmetic, look into fixing the `pylint` issues as well (looks like deps are not installed correctly)

What's the command to see which statement line numbers were missed?

**Comment by leonkozlowski** [comment]
> > Mostly cosmetic, look into fixing the `pylint` issues as well (looks like deps are not installed correctly)
> 
> What's the command to see which statement line numbers were missed?

```
betfund_solicitor/ses.py:83:0: C0304: Final newline missing (missing-final-newline)
betfund_solicitor/ses.py:1:0: C0114: Missing module docstring (missing-module-docstring)
betfund_solicitor/ses.py:3:0: E0401: Unable to import 'boto3' (import-error)
betfund_solicitor/ses.py:4:0: E0401: Unable to import 'botocore.exceptions' (import-error)
```
Traceback above

Add `boto3` to `setup.py` should fix it I suppose

**Review by leonkozlowski — APPROVED** [review]
Should this go to master or development?

**Comment by mitchbregs** [comment]
> Should this go to master or development?

Good catch. Changed base.

**Comment by leonkozlowski** [comment]
`betfund_solicitor/message.py:96:0: C0305: Trailing newlines (trailing-newlines)`

**Comment by mitchbregs** [comment]
> `betfund_solicitor/message.py:96:0: C0305: Trailing newlines (trailing-newlines)`

Rebased to 1 commit, fixed `pylint`

**Review by leonkozlowski — APPROVED** [review]
good to go

**Pull request merged**